src/connectors/CassandraConnector.js

```diff
@@ -16,22 +25,35 @@ module.exports = {
     const client = new cassandra.Client(options);
 
     return new Promise((resolve, reject) => {
+      let start = Date.now();
       client.connect()
         .then(() => {
+          let duration = Date.now() - start;
+          iclient.trackMetric('Cassandra connection time', duration);
           resolve(client);
         })
         .catch((err) => {
+          iclient.trackException(err);
           client.shutdown();
           reject(err);
         });
     });
   },
 
   closeClient: client => {
+    let start = Date.now();
     return new Promise((resolve, reject) => {
       client.shutdown()
-        .then(resolve)
-        .catch(reject);
+        .then(() => {
+          let duration = Date.now() - start;
+          iclient.trackMetric('Cassandra shutdown time', duration);
```

Comment:
I wonder if we can make this more generic/re-usable. Pseudo-code:

```js
withMetric('Cassandra connection time', () => new Promise((resolve, reject) => {
  client.connect()
  .catch((err) => {
    client.shutdown();
	reject(err);
  });
});

function withMetric(metricName, promiseAction) {
  let start = Date.now();
  promiseAction().then((val) => {
    let duration = Date.now() - start;
	iclient.trackMetric(metricName, duration);
	resolve(val);
  })
  .catch((err) => {
    iclient.trackException(err)
	reject(err);
  });
}
```

Adding a slight level of abstraction on top of using the naked client also has the advantage of separating the logging implementation from the code that does the business logic. Here, everything is still fine, but it's a slippery slope and easy to get to the situation where the logging/argument-checking code takes up the majority of a method and drowns out the business logic parts. Using this decorator style to handle logging will set a good example to prevent that.

Maybe it's worth spending 15 minutes to see if we can factor out the logging code like this, but it's not a high priority item so don't worry if you run into a roadblock.